### ISSUE TITLE
Logging intermediate/cumulative  rewards

### ISSUE THREAD

**Issue by Driesssens** [lcswillems/torch-rl#15]
Although MiniGrid standardly uses a single discounted reward at the end of an episode, I want to experiment with intermediate rewards. Specifically, I use a RewardWrapper that replaces the discounted final reward with a non-discounted final reward (of 1), and instead yields a small negative reward at every step to encourage faster completion.

I don't see their effect on the logged values. Browsing through the code, I encountered in `BaseAlgo`:

```self.mask = 1 - torch.tensor(done, device=self.device, dtype=torch.float)```

and
```
self.log_episode_return *= self.mask
self.log_episode_reshaped_return *= self.mask
self.log_episode_num_frames *= self.mask
```

Do I understand correctly that this is to filter out non-final steps from the logged statistics - a trick that works if all rewards are issued at the final step of episodes, but doesn't work if we want statistics on the cumulative rewards during one episode? I'd imagine that fixing that is non-trivial, as endings of episodes do not coincide with model update steps and we would need a way to count the cumulative rewards per process asynchronously with the training loop?

Also, do I understand correctly that this only affects the logs and that the intermediate rewards are still taken into account during training?

**Comment by lcswillems** [comment]
I don't have enough informations to help you debug. What is precisely the `reshape_reward` function? Can you show what are you logs?

I don't think the problem is coming from the code you highlight. This code just updates the logs at the end of the episode, but it doesn't change the return.

**Comment by lcswillems** [comment]
Hi,

Did you solve your problem? If yes, could you explain me what was wrong in my code?
I will close the issue in few days if I don't have additional informations.

Thank you,
Lucas

**Comment by lcswillems** [comment]
I close your issue because I don't have enough informations. I will reopen it if I know more.
